hypervisor/boot/multiboot.c

```diff
@@ -41,29 +41,34 @@ void init_acrn_multiboot_info(void)
 
 int32_t sanitize_multiboot_info(void)
 {
-	uint32_t mmap_entry_size = 0U;
-
-	if (boot_from_multiboot1()) {
-		pr_info("Multiboot1 detected.");
-		mmap_entry_size = sizeof(struct multiboot_mmap);
-#ifdef CONFIG_MULTIBOOT2
-	} else if (boot_from_multiboot2()) {
-		pr_info("Multiboot2 detected.");
-		mmap_entry_size = sizeof(struct multiboot2_mmap_entry);
-#endif
-	} else {
-		/* mbi_status is still -ENODEV, nothing to do here */
-	}
-
 	if ((acrn_mbi.mi_mmap_entries != 0U) && (acrn_mbi.mi_mmap_va != NULL)) {
 		if (acrn_mbi.mi_mmap_entries > E820_MAX_ENTRIES) {
 			pr_err("Too many E820 entries %d\n", acrn_mbi.mi_mmap_entries);
 			acrn_mbi.mi_mmap_entries = E820_MAX_ENTRIES;
 		}
-		(void)memcpy_s((void *)(&acrn_mbi.mi_mmap_entry[0]),
-			(acrn_mbi.mi_mmap_entries * mmap_entry_size),
-			(const void *)acrn_mbi.mi_mmap_va,
-			(acrn_mbi.mi_mmap_entries * mmap_entry_size));
+		if (boot_from_multiboot1()) {
+			uint32_t mmap_entry_size = sizeof(struct multiboot_mmap);
+
+			pr_info("Multiboot1 detected.");
```

Comment:
do we need seperate pr_info? 1:this is very easy to know from boot cmd line, 2: even if we want to print, we can print boot_regs[0]， or we can print all boot_regs somewhere.

Reply:
@dongyaozu 
updated, please check.